### PR TITLE
Bug 1677559 - Run visual metrics processing within the browsertime test task.

### PR DESCRIPTION
**Pull request by gmierz**
This patch removes the `-vismet` browsertime tasks from CI and instead runs visual-metrics processing within the browsertime test task.

This will need to land after the patch series in bug 1677559 lands: https://bugzilla.mozilla.org/show_bug.cgi?id=1677559

### CODE REVIEW

**Comment by gmierz** [comment]
I should mention that the browsertime tests are likely to fail this time since it requires some changes in mozilla-central which doesn't exist yet.

**Comment by gmierz** [comment]
The patch series in https://bugzilla.mozilla.org/show_bug.cgi?id=1677559 was pushed to autoland. Once it hits mozilla-central, I'll retrigger the tests here.

File: taskcluster/ci/browsertime/kind.yml [hunk 1]
```diff
@@ -37,7 +37,7 @@ job-defaults:
                     subject: '[{product_name}] Raptor-Browsertime job "{task_name}" failed'
                     to-addresses: [perftest-alerts@mozilla.com]
             default: {}
-    run-on-tasks-for: []
+    run-on-tasks-for: [github-pull-request]
```
Comment (escapewindow) [inline taskcluster/ci/browsertime/kind.yml:40]: If I run `TASK_ID=foo taskgraph target-graph --diff HEAD~1 -p taskcluster/test/params/main-repo-pull-request.yml -J` I get a ton of changes that seem unrelated. Namely, we pull in nightly builds into pull request graphs; I think that's unwanted; we can see those tasks run against this PR. I suspect this may be the line responsible.

Comment (gmierz) [reply taskcluster/ci/browsertime/kind.yml:40]: Ah yes, sorry about that! I added this to run the browsertime tests in the PR to see how they would go. I'll re-run the tests again once the patch hits m-c.

**Comment by gmierz** [comment]
I've retriggered the tests.

**Comment by gmierz** [comment]
I'm going to undo the testing changes now, here's a link to the task group running the browsertime tests: https://firefox-ci-tc.services.mozilla.com/tasks/axCKuk97REqXvlYlCexPCw

**Comment by gmierz** [comment]
@escapewindow the tests are passing now (see comment above for a link to the group), and the patch is ready for a review now.

Here's a direct link to one of the logs (no more node related failures): https://firefoxci-websocktunnel.services.mozilla.com/bitbar.motog5-22.60099/log/CU2nrlBUTwycOOwoiLXZ9g

Along with an excerpt showing that visual metrics are being processed within the task:
```
[task 2022-05-03T12:32:24.611Z] 12:29:45     INFO -  raptor-browsertime Info: Navigating to about:blank, count: 1
[task 2022-05-03T12:32:24.611Z] 12:29:45     INFO -  raptor-browsertime Info: Navigating to url about:blank iteration 1
[task 2022-05-03T12:32:24.611Z] 12:29:51     INFO -  raptor-browsertime Info: Navigating to primary url:https://www.booking.com/
[task 2022-05-03T12:32:24.611Z] 12:29:51     INFO -  raptor-browsertime Info: Cycle 1, waiting for 1000 ms
[task 2022-05-03T12:32:24.611Z] 12:29:52     INFO -  raptor-browsertime Info: Cycle 1, starting the measure
[task 2022-05-03T12:32:24.611Z] 12:29:52     INFO -  raptor-browsertime Info: Testing url https://www.booking.com/ iteration 1
[task 2022-05-03T12:32:24.611Z] 12:29:52     INFO -  raptor-browsertime Info: Tue, 03 May 2022 12:29:52 GMT adb:command Send '0019host:transport:ZY3223LVXH'
[task 2022-05-03T12:32:24.611Z] 12:29:52     INFO -  raptor-browsertime Info: Tue, 03 May 2022 12:29:52 GMT adb:command Send '003Dshell:screenrecord --bit-rate 8000000 /sdcard/browsertime.mp4'
[task 2022-05-03T12:32:24.611Z] 12:30:01     INFO -  raptor-browsertime Info: https://www.booking.com/ has been tested before within the same run, it will get an extra query parameter named browsertime_run. Make sure to use alias to keep track of the URLs
[task 2022-05-03T12:32:24.611Z] 12:30:01     INFO -  raptor-browsertime Info: Tue, 03 May 2022 12:30:01 GMT adb:command Send '0019host:transport:ZY3223LVXH'
[task 2022-05-03T12:32:24.611Z] 12:30:01     INFO -  raptor-browsertime Info: Tue, 03 May 2022 12:30:01 GMT adb:command Send '0088shell:command -v pkill >/dev/null && pkill -l SIGINT screenrecord || kill -2 $(ps screenrecord | grep -Eo [0-9]+ | grep -m 1 -Eo [0-9]+)'
[task 2022-05-03T12:32:24.611Z] 12:30:06     INFO -  raptor-browsertime Info: Tue, 03 May 2022 12:30:06 GMT adb:command Send '0019host:transport:ZY3223LVXH'
[task 2022-05-03T12:32:24.611Z] 12:30:06     INFO -  raptor-browsertime Info: Tue, 03 May 2022 12:30:06 GMT adb:command Send '0005sync:'
[task 2022-05-03T12:32:24.611Z] 12:30:06     INFO -  raptor-browsertime Info: Tue, 03 May 2022 12:30:06 GMT adb:sync RECV /sdcard/browsertime.mp4
[task 2022-05-03T12:32:24.611Z] 12:30:07     INFO -  raptor-browsertime Info: Tue, 03 May 2022 12:30:07 GMT adb:command Send '0019host:transport:ZY3223LVXH'
[task 2022-05-03T12:32:24.611Z] 12:30:07     INFO -  raptor-browsertime Info: Tue, 03 May 2022 12:30:07 GMT adb:command Send '002Dshell:rm -- "/sdcard//sdcard/browsertime.mp4"'
[task 2022-05-03T12:32:24.611Z] 12:30:08     INFO -  raptor-browsertime Info: Browsertime pageload ended.
[task 2022-05-03T12:32:24.611Z] 12:30:12     INFO -  raptor-browsertime Info: Tue, 03 May 2022 12:30:12 GMT adb:command Send '0019host:transport:ZY3223LVXH'
[task 2022-05-03T12:32:24.611Z] 12:30:12     INFO -  raptor-browsertime Info: Tue, 03 May 2022 12:30:12 GMT adb:command Send '0040shell:dumpsys battery | grep temperature | grep -Eo '[0-9]{1,3}''
[task 2022-05-03T12:32:24.611Z] 12:30:12     INFO -  raptor-browsertime Info: Get visual metrics from the video
[task 2022-05-03T12:32:24.611Z] 12:30:19     INFO -  raptor-browsertime Info: Converting video to 60 fps
[task 2022-05-03T12:32:24.611Z] 12:30:28     INFO -  raptor-browsertime Info: Get visual metrics from the video
[task 2022-05-03T12:32:24.611Z] 12:30:33     INFO -  raptor-browsertime Info: Converting video to 60 fps
[task 2022-05-03T12:32:24.611Z] 12:30:47     INFO -  raptor-browsertime Info: https://www.booking.com/ TTFB: 35ms DOMContentLoaded: 1.69s firstPaint: 1.54s FCP: 1.16s Load: 2.08s
[task 2022-05-03T12:32:24.611Z] 12:30:47     INFO -  raptor-browsertime Info: VisualMetrics: FirstVisualChange: 1.64s SpeedIndex: 2.67s PerceptualSpeedIndex: 3.26s ContentfulSpeedIndex: 2.07s VisualComplete85: 3.82s LastVisualChange: 4.76s
[task 2022-05-03T12:32:24.611Z] 12:30:47     INFO -  raptor-browsertime Info: https://www.booking.com/?browsertime_run=2 TTFB: 33ms DOMContentLoaded: 1.16s firstPaint: 661ms FCP: 511ms Load: 1.35s
[task 2022-05-03T12:32:24.611Z] 12:30:47     INFO -  raptor-browsertime Info: VisualMetrics: FirstVisualChange: 717ms SpeedIndex: 1.71s PerceptualSpeedIndex: 2.16s ContentfulSpeedIndex: 1.14s VisualComplete85: 2.59s LastVisualChange: 3.54s
```

**Comment by gmierz** [comment]
Failures on the G5 are all the same and seem unrelated to the patch:
```
[task 2022-05-03T12:50:45.116Z] 12:50:45    FATAL - Uncaught exception: Traceback (most recent call last):
[task 2022-05-03T12:50:45.116Z] 12:50:45    FATAL -   File "/builds/task_165158209025049/workspace/mozharness/mozharness/base/script.py", line 2358, in run
[task 2022-05-03T12:50:45.116Z] 12:50:45    FATAL -     self.run_action(action)
[task 2022-05-03T12:50:45.116Z] 12:50:45    FATAL -   File "/builds/task_165158209025049/workspace/mozharness/mozharness/base/script.py", line 2292, in run_action
[task 2022-05-03T12:50:45.116Z] 12:50:45    FATAL -     self._possibly_run_method(method_name, error_if_missing=True)
[task 2022-05-03T12:50:45.116Z] 12:50:45    FATAL -   File "/builds/task_165158209025049/workspace/mozharness/mozharness/base/script.py", line 2244, in _possibly_run_method
[task 2022-05-03T12:50:45.116Z] 12:50:45    FATAL -     return getattr(self, method_name)()
[task 2022-05-03T12:50:45.116Z] 12:50:45    FATAL -   File "/builds/task_165158209025049/workspace/mozharness/mozharness/mozilla/testing/raptor.py", line 1140, in install
[task 2022-05-03T12:50:45.116Z] 12:50:45    FATAL -     self.device.uninstall_app(self.binary_path)
[task 2022-05-03T12:50:45.116Z] 12:50:45    FATAL -   File "/builds/task_165158209025049/workspace/build/venv/lib/python3.6/site-packages/mozdevice/adb.py", line 4385, in uninstall_app
[task 2022-05-03T12:50:45.116Z] 12:50:45    FATAL -     if self.is_app_installed(app_name, timeout=timeout):
[task 2022-05-03T12:50:45.116Z] 12:50:45    FATAL -   File "/builds/task_165158209025049/workspace/build/venv/lib/python3.6/site-packages/mozdevice/adb.py", line 4044, in is_app_installed
[task 2022-05-03T12:50:45.116Z] 12:50:45    FATAL -     "pm list package %s" % app_name, timeout=timeout, enable_run_as=False
[task 2022-05-03T12:50:45.116Z] 12:50:45    FATAL -   File "/builds/task_165158209025049/workspace/build/venv/lib/python3.6/site-packages/mozdevice/adb.py", line 2222, in shell_output
[task 2022-05-03T12:50:45.116Z] 12:50:45    FATAL -     raise ADBProcessError(adb_process)
[task 2022-05-03T12:50:45.116Z] 12:50:45    FATAL - mozdevice.adb.ADBProcessError: args: adb wait-for-device shell su -c 'pm list package org.mozilla.fenix', exitcode: 255, stdout: android.os.DeadObjectException: Transaction failed on small parcel; remote process probably died
[task 2022-05-03T12:50:45.116Z] 12:50:45    FATAL - 	at android.os.BinderProxy.transactNative(Native Method)
[task 2022-05-03T12:50:45.116Z] 12:50:45    FATAL - 	at android.os.BinderProxy.transact(Binder.java:615)
[task 2022-05-03T12:50:45.116Z] 12:50:45    FATAL - 	at android.os.BinderProxy.shellCommand(Binder.java:662)
[task 2022-05-03T12:50:45.116Z] 12:50:45    FATAL - 	at com.android.commands.pm.Pm.runShellCommand(Pm.java:291)
[task 2022-05-03T12:50:45.116Z] 12:50:45    FATAL - 	at com.android.commands.pm.Pm.runList(Pm.java:640)
[task 2022-05-03T12:50:45.116Z] 12:50:45    FATAL - 	at com.android.commands.pm.Pm.run(Pm.java:126)
[task 2022-05-03T12:50:45.116Z] 12:50:45    FATAL - 	at com.android.commands.pm.Pm.main(Pm.java:95)
[task 2022-05-03T12:50:45.116Z] 12:50:45    FATAL - 	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
[task 2022-05-03T12:50:45.116Z] 12:50:45    FATAL - 	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:262)
[task 2022-05-03T12:50:45.116Z] 12:50:45    FATAL - Running post_fatal callback...
```

**Review by escapewindow — APPROVED** [review]
Thank you!
